examples/routing_service/udp_socket_adapter/src/UdpSocket.hpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * (c) 2019 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+ *
+ * RTI grants Licensee a license to use, modify, compile, and create derivative
+ * works of the Software.  Licensee has the right to distribute object form
+ * only for use with RTI products.  The Software is provided "as is", with no
+ * warranty of any type, including any warranty for fitness for any purpose.
+ * RTI is under no obligation to maintain or support the Software.  RTI shall
+ * not be liable for any incidental or consequential damages arising out of the
+ * use or inability to use the software.
+ */
+
+#ifdef _WIN32
+    #include <winsock2.h>
+    #include <ws2tcpip.h>
+#else
+    #include <sys/socket.h>
+    #include <netinet/in.h>
+    #include <arpa/inet.h>
+    #include <unistd.h>
+    #include <fcntl.h>
+    #include <cstring>
+#endif
+
+#include <iostream>
+
+#ifdef _WIN32
+    #pragma comment(lib, "ws2_32.lib")
+#endif
+
+class UdpSocket {
+public:
+    UdpSocket(const char* ip, int port);
+    ~UdpSocket();
+    void receive_data(
+            char* received_buffer,
+            int* received_bytes,
+            int size_of_original_buffer);
+
+private:
+#ifdef _WIN32
```

Comment:
It's a pity that the use of our Osapi layer is discouraged in examples, because it would provide a way to abstract from all the different OS-specific nuances.

Reply:
I agree! It would've saved me some work here